src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotDisableSchUseStrongCryptoTests.cs

```diff
@@ -0,0 +1,182 @@
+{
+    public void TestMethod()
+    {
+        AppContext.SetSwitch(""Switch.System.Net.DontEnableSchUseStrongCrypto"", true);
+    }
+}",
+            GetCSharpResultAt(8, 9, DoNotDisableSchUseStrongCrypto.Rule, "SetSwitch"));
+        }
+
+        [Fact]
+        public void TestEquationDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        AppContext.SetSwitch(""Switch.System.Net.DontEnableSchUseStrongCrypto"", 1 + 2 == 3);
+    }
+}",
+            GetCSharpResultAt(8, 9, DoNotDisableSchUseStrongCrypto.Rule, "SetSwitch"));
+        }
+
+        [Fact]
+        public void TestConditionalOperatorDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        AppContext.SetSwitch(""Switch.System.Net.DontEnableSchUseStrongCrypto"", 1 == 1 ? true : false);
+    }
+}",
+            GetCSharpResultAt(8, 9, DoNotDisableSchUseStrongCrypto.Rule, "SetSwitch"));
+        }
+
+        [Fact]
+        public void TestWithConstantSwitchNameDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        const string constSwitchName = ""Switch.System.Net.DontEnableSchUseStrongCrypto"";
+        AppContext.SetSwitch(constSwitchName, true);
+    }
+}",
+            GetCSharpResultAt(9, 9, DoNotDisableSchUseStrongCrypto.Rule, "SetSwitch"));
+        }
+
+        [Fact]
+        public void TestBoolNoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        AppContext.SetSwitch(""Switch.System.Net.DontEnableSchUseStrongCrypto"", false);
+    }
+}");
+        }
+
+        [Fact]
+        public void TestEquationNoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        AppContext.SetSwitch(""Switch.System.Net.DontEnableSchUseStrongCrypto"", 1 + 2 != 3);
+    }
+}");
+        }
+
+        [Fact]
+        public void TestConditionalOperatorNoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        AppContext.SetSwitch(""Switch.System.Net.DontEnableSchUseStrongCrypto"", 1 == 1 ? false : true);
+    }
+}");
+        }
+
+        [Fact]
+        public void TestSwitchNameNullNoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        AppContext.SetSwitch(null, true);
+    }
+}");
+        }
+
+        [Fact]
+        public void TestSwitchNameVariableNoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        string switchName = ""Switch.System.Net.DontEnableSchUseStrongCrypto"";
+        AppContext.SetSwitch(switchName, true);
+    }
+}");
+        }
```

Comment:
Leave a comment noting that ideally, we would generate a diagnostic in this case.